DeliveryApi/cdk/src/main/java/com/ilmlf/delivery/db/DbStack.java

```diff
@@ -0,0 +1,249 @@
+     * in case of failover. By default this stack deploys a database instance and its replica to different AZs.
+     */
+
+    // The `Vpc` construct creates subnets for you automatically
+    // See https://docs.aws.amazon.com/cdk/api/latest/docs/aws-ec2-readme.html#vpc for details
+    this.vpc = new Vpc(this, "farmer-Vpc");
+
+    // Security group acts as a virtual firewall to control inbound and outbound traffic
+    this.securityGroup =
+        new SecurityGroup(
+            this,
+            "FarmerDeliverySG",
+            SecurityGroupProps.builder()
+                .vpc(vpc)
+                .description("Shared SG for database and proxy")
+                .allowAllOutbound(true)
+                .build());
+
+    /**
+     * #################
+     * ### DB Instance #
+     * #################
+     * Creates a MYSQL RDS instance.
+     *
+     * This construct also creates a secret store in AWS Secrets Manager. You can retrieve
+     * reference to the secret store by calling farmerDB.getSecret()
+     *
+     * The secret store contains the admin username, password and other DB information for connecting to the DB
+     *
+     * For production, consider using `DatabaseCluster` to create multiple instances in different AZs.
+     * This costs more but you will have higher availability.
+     *
+     * See https://docs.aws.amazon.com/cdk/api/latest/docs/aws-rds-readme.html for details.
+     */
+    String dbName = "FarmerDB";
+
+    List<ISubnet> subnets = "public".equals(this.getNode().tryGetContext("subnetType")) ?
+        vpc.getPublicSubnets() :
+        vpc.getPrivateSubnets();
+
+    DatabaseInstance farmerDb =
+        new DatabaseInstance(
+            this,
+            dbName,
+            DatabaseInstanceProps.builder()
+                .vpc(vpc)
+                .securityGroups(List.of(securityGroup))
+                // Using MySQL engine
+                .engine(
+                    DatabaseInstanceEngine.mysql(
+                        MySqlInstanceEngineProps.builder()
+                            .version(MysqlEngineVersion.VER_5_7_31)
+                            .build()))
+                .instanceType(InstanceType.of(InstanceClass.BURSTABLE2, InstanceSize.SMALL))
+                .vpcSubnets(SubnetSelection.builder().subnets(subnets).build())
+                .storageEncrypted(true)
+                .multiAz(true)
+                .autoMinorVersionUpgrade(true)
+                .allocatedStorage(25)
+                .publiclyAccessible(true)
+                .storageType(StorageType.GP2)
+                .backupRetention(Duration.days(7))
+                .deletionProtection(false)
+                // Create an admin credential for connectin to database. This credential will
+                // be stored in a Secret Manager store.
+                .credentials(Credentials.fromGeneratedSecret(dbName + "admin"))
+                .databaseName(dbName)
+                .port(3306)
```

Comment:
We should avoid hardcoded db port and put it in a context variable

Reply:
dbPort context var added, if none found resorts to 3306

Reply:
Will do.